api/src/main/java/me/lokka30/treasury/api/economy/account/Account.java

```diff
@@ -413,6 +419,49 @@ CompletableFuture<Response<TriState>> setPermission(
      */
     CompletableFuture<Response<Map<AccountPermission, TriState>>> retrievePermissions(@NotNull UUID player);
 
+    /**
+     * Returns a {@link Map} abomination, with the permissions of each account member.
+     * <p><b>Strongly advised not to leave the default implementation of this method.
+     * Although it works, it may make an awfully lot of requests.</b>
+     *
+     * @return future which if successful returns a map of member id as a key and permissions as
+     *         values.
+     * @since v2.0.0
+     */
+    @NotNull
+    default CompletableFuture<Response<Map<UUID, Set<Map.Entry<AccountPermission, TriState>>>>> retrievePermissionsMap() {
+        return this.retrieveMemberIds().thenComposeAsync(resp -> {
+            if (!resp.isSuccessful()) {
+                return CompletableFuture.completedFuture(Response.failure(resp.getFailureReason()));
+            }
+            List<UUID> ids = new LinkedList<>(resp.getResult());
+            Queue<CompletableFuture<Response<Map<AccountPermission, TriState>>>> fifo = new ArrayDeque<>();
+            for (UUID uuid : ids) {
+                fifo.add(this.retrievePermissions(uuid));
+            }
+            return CompletableFuture
+                    .allOf(fifo.toArray(new CompletableFuture[0]))
+                    .thenApply(ignored -> {
+                        Map<UUID, Set<Map.Entry<AccountPermission, TriState>>> ret = new ConcurrentHashMap<>();
+                        for (UUID uuid : ids) {
+                            CompletableFuture<Response<Map<AccountPermission, TriState>>> future = fifo.poll();
+                            if (future == null) {
+                                throw new RuntimeException("Queue ended but there are still ids?");
+                            }
+                            // safe to call .join - since it's all been done
+                            Response<Map<AccountPermission, TriState>> response = future.join();
+                            if (!response.isSuccessful()) {
+                                // TODO: this is swallowing potential failures - METHOD FOR
+                                //  LOGGING NEEDED
```

Comment:
any ideas??

Reply:
If logging errors here is difficult and the default implementation is not advisable due to how many requests it makes, should we just not have a default implementation? Alternately, would failing to fetch data for one UUID be considered a full failure because it is not a complete set of permission mappings?